### PR TITLE
Birmiwal.rahul development

### PR DESCRIPTION
**Pull request by RBirmiwal**
This branch contains updates that incorporate the following desired tasks: 

1. Automate docs build from master to gh-pages 
2. Unittesting for problem and system

These files are test_system.py, test_problem.py, .github/workflows/update_docs.yml, /docs/readme.md

There is an update to the readme in /docs to reflect the change in documentation update. 
There is also a one-line change to the docstring of System's MovingHorizon class (system.py) -- correcting an error that the output should be a 3D tensor of shape (ndelay, batch, dim)

### CODE REVIEW

**Comment by madelynshapiro** [comment]
Is the `tmp/html/` folder necessary to merge into master for the automated gh-pages build or is it leftover from testing the docs deployment?

**Review by drgona — APPROVED** [review]
nicely done!

**Comment by RBirmiwal** [comment]
Made some additional changes to test_system.py and system.py. Added in output correctness check to MovingHorizon forward. Also modified associated docstring in system.py. The output should indeed be of dimension [ndelay, batch, data_dim]. One can confirm this by running: 

```
torch.manual_seed(0)

# 1, wrap nn.Linear into Node
net_1 = lambda x: x*2
node_1 = Node(net_1, ['x1'], ['y1'])
# print input and output keys
# evaluate forward pass of the node with dictionary input dataset
input_data = {'x1': torch.rand(3,1)}
output = (node_1(input_data))

horizon = MovingHorizon(node_1, ndelay=5)
baz = horizon(input_data)

print(baz['y1'].shape) 
```
should print torch.size([5,3,1])
